File: debop4s-data/src/main/scala/com/github/debop4s/data/model/LongEntity.scala
package com.github.debop4s.data.model

import javax.persistence.{MappedSuperclass, Id, GeneratedValue}

/**
 * Created by debop on 2014. 3. 8.
 */
@MappedSuperclass
abstract class LongEntity extends HibernateEntity[java.lang.Long] {

  @Id
  @GeneratedValue
  var id: java.lang.Long = _

  def getId = id

}

File: debop4s-data-orm/src/main/java/debop4s/data/orm/model/LongEntity.java
package debop4s.data.orm.model;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;


@MappedSuperclass
@DynamicInsert
@DynamicUpdate
@Getter
public class LongEntity extends HibernateEntityBase<Long> {

    @Id
    @GeneratedValue
    @Setter(AccessLevel.PROTECTED)
    private Long id;

    private static final long serialVersionUID = 8966803355321799991L;

}
